schemast/index.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019-present Facebook
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schemast
+
+import (
+	"go/ast"
+
+	"entgo.io/ent"
+	"entgo.io/ent/schema/index"
+)
+
+// Index converts a *index.Descriptor back into an *ast.CallExpr of the ent Index package that can be used
+// to construct it.
+func Index(desc *index.Descriptor) (*ast.CallExpr, error) {
+	idx := newIndexCall(desc)
+	if desc.Unique {
+		idx.method("Unique")
+	}
+	if desc.StorageKey != "" {
+		idx.method("StorageKey", strLit(desc.StorageKey))
+	}
+	if len(desc.Edges) > 0 {
+		var edges []ast.Expr
+		for _, edg := range desc.Edges {
+			edges = append(edges, strLit(edg))
+		}
+		idx.method("Edges", edges...)
```

Comment:
```suggestion
		var edges []ast.Expr
		for _, e := range desc.Edges {
			edges = append(edges, strLit(e))
		}
		idx.method("Edges", edges...)
```

You can use `make` btw.